Assets/Dependencies/SmashBrew/Stage/StageLedge.cs

```diff
@@ -15,12 +15,24 @@ public class StageLedge : MonoBehaviour {
             if (!collider.CompareTag(Config.Tags.LedgeTag))
                 return;
             var movement = collider.GetComponentInParent<MovementState>();
-            if (movement == null)
+            if (movement == null || Occupied)
```

Comment:
This check should be inside Grab itself. It's a fundamental requirement: that the ledge is not grabbed yet.